pants-plugins/src/python/internal_backend/utilities/register.py

```diff
@@ -211,20 +214,21 @@ def __init__(
 
         super().__init__(address, payload, provides=setup_py, **kwargs)
 
-        if build_file_aliases or register_goals or global_subsystems or rules:
+        if build_file_aliases or register_goals or global_subsystems or rules or target_types:
             module = os.path.relpath(address.spec_path, self.target_base).replace(os.sep, ".")
-            entrypoints = []
+            entry_points = []
             if build_file_aliases:
-                entrypoints.append(f"build_file_aliases = {module}.register:build_file_aliases")
+                entry_points.append(f"build_file_aliases = {module}.register:build_file_aliases")
             if register_goals:
-                entrypoints.append(f"register_goals = {module}.register:register_goals")
+                entry_points.append(f"register_goals = {module}.register:register_goals")
             if global_subsystems:
-                entrypoints.append(f"global_subsystems = {module}.register:global_subsystems")
+                entry_points.append(f"global_subsystems = {module}.register:global_subsystems")
             if rules:
-                entrypoints.append(f"rules = {module}.register:rules")
-            entry_points = {"pantsbuild.plugin": entrypoints}
+                entry_points.append(f"rules = {module}.register:rules")
+            if target_types:
+                entry_points.append(f"target_types = {module}.register:target_types")
 
-            setup_py.setup_py_keywords["entry_points"] = entry_points
+            setup_py.setup_py_keywords["entry_points"] = {"pantsbuild.plugin": entry_points}
```

Comment:
Is this something that setup_py was consuming under this name, or is it only for our consumption?

https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html#the-console-scripts-entry-point

Reply:
I'm not sure - this is the same as before, just now inlined.

Reply:
Ooh, I see.